lib/TransactionRunner.js

```diff
@@ -505,13 +505,6 @@ Not performing HTTP request for '${transaction.name}'.\
       }
       transaction.real = real;
 
-      if (!transaction.real.body && transaction.expected.body) {
-        // Leaving body as undefined skips its validation completely. In case
-        // there is no real body, but there is one expected, the empty string
-        // ensures Gavel does the validation.
-        transaction.real.body = '';
```

Comment:
I assumed there is no need to coerce this since it will be done by Gavel.
However, gavel has no effect over the `transaction.real` or `transaction.expected` after the execution. I would like somebody to verify that this removal is safe.

It _is_ safe if `transactions.real` is used only as an input to `gavel.validate()`, and is not expected to be mutated by Gavel, or relied directly (like in test suites to assert `transaction.real.body`).

Reply:
This is an interesting question. I think we want to prevent mutation in our new code, but the fact is that how Dredd is designed now is that the transaction object is supposed to be mutated on the way. So if we don't mutate the object, we might break e.g. all `after` hooks, where users could rely on the transaction object being updated with this (and other) coercions. I'd probably keep the mutation for now.

There is a plan to redesign the transaction object to better distinguish things which should not get mutated and things which can be mutated e.g. by hooks etc. For example now there is just "request" and it is not clear whether it is (expected) request as was described in the API description, or whether it is (real) request as sent over the wire. In fact currently it is both.

Reply:
Hmm. Honestly, I don't know. This was definitely made to hack Gavel into desired behavior. If Gavel does it itself, perhaps we can just remove it.